dev-test/gql-tag-operations/graphql/fragment-masking.ts

```diff
@@ -1,48 +1,46 @@
-import { ResultOf, TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
+import { ResultOf, DocumentTypeDecoration } from '@graphql-typed-document-node/core';
 
-export type FragmentType<TDocumentType extends DocumentNode<any, any>> = TDocumentType extends DocumentNode<
-  infer TType,
-  any
->
-  ? TType extends { ' $fragmentName'?: infer TKey }
-    ? TKey extends string
-      ? { ' $fragmentRefs'?: { [key in TKey]: TType } }
+export type FragmentType<TDocumentType extends DocumentTypeDecoration<any, any>> =
+  TDocumentType extends DocumentTypeDecoration<infer TType, any>
+    ? TType extends { ' $fragmentName'?: infer TKey }
+      ? TKey extends string
+        ? { ' $fragmentRefs'?: { [key in TKey]: TType } }
+        : never
```

Comment:
Is this considered a breaking change for the codegen or not? @n1ru4l @saihaj

Reply:
I don't think so, people shouldn't use this type with anything else than our generated stuff.

Reply:
agreed should be fine

Reply:
Okay, as I think about it a bit more, this change kinda requires people to upgrade their typed document node version. 🤔 

In graphql-yoga we treat `peerDependency` upgrades depending on the type of change.

e.g. 
`X.1.1` bump of a peer dependency would be a breaking change
`1.X.1` bump of a peer dependency would be a minor change
`1.1.X` bump of a peer dependency would be a patch change

So I think we should treat this the same here (minor change).

Here the special case is that it is not listed as a peer dependency, but theoretically, it could be one, as people will need to install it, otherwise, their code won't compile with strict package manages like pnpm and yarn that do no hoisting. 🤔 

Then, however, whether it should be a peer dependency or not also depends on the location where the code is generated.

So the safest option would be to ship this as a breaking change. I am sure people will open issues if we don't...